src/engraving/draw/qpainterprovider.cpp

```diff
@@ -132,19 +132,21 @@ const Font& QPainterProvider::font() const
     return m_font;
 }
 
-void QPainterProvider::setPen(const QPen& pen)
+void QPainterProvider::setPen(const Pen& pen)
 {
-    m_painter->setPen(pen);
+    m_pen = pen;
+    m_painter->setPen(m_pen.toQPen());
```

Comment:
If we introduce a member `m_pen`, then it needs to be initialized with a pen from the painter, as is done for the font.
But I think in this case we don't need to add a class member, it's enough just to do in the getter
```
const Pen& QPainterProvider::pen() const
{
    return Pen::fromQPen(m_painter->pen());
}
```

Reply:
@igorkorsukov then we have a reference to a local object, i will initialize m_pen from the painter